src/plugins/intel_cpu/src/nodes/experimental_detectron_roifeatureextractor.cpp

```diff
@@ -346,66 +183,244 @@ void ExperimentalDetectronROIFeatureExtractor::initSupportedPrimitiveDescriptors
+        impl_type = impl_desc_type::jit_sse42;
+    } else {
+        impl_type = impl_desc_type::ref;
+    }
 
     addSupportedPrimDesc(inDataConf,
-                         {{LayoutType::ncsp, Precision::FP32},
+                         {{LayoutType::ncsp, outputPrec},
                           {LayoutType::ncsp, Precision::FP32}},
-                         impl_desc_type::ref_any);
+                         impl_type);
+}
+
+void ExperimentalDetectronROIFeatureExtractor::createPrimitive() {
+    Node::createPrimitive();
+
+    if (!roi_align_kernel_) {
+        createJitKernel(getParentEdgeAt(INPUT_FEATURES_START)->getMemoryPtr()->getDesc().getPrecision(), ROIAlignLayoutType::ncsp);
+    }
+}
+
+void ExperimentalDetectronROIFeatureExtractor::createJitKernel(const InferenceEngine::Precision& dataPrec, const ROIAlignLayoutType& selectLayout) {
+    auto jcp = jit_roi_align_params();
+    jcp.alg = Algorithm::ROIAlignAvg;
+    jcp.data_prc = dataPrec;
+    jcp.data_size = dataPrec.size();
+    jcp.layout = selectLayout;
+    jcp.pooled_h = pooled_height_;
+    jcp.pooled_w = pooled_width_;
+
+    if (mayiuse(cpu::x64::avx512_core)) {
+        roi_align_kernel_.reset(new jit_uni_roi_align_kernel_f32<cpu::x64::avx512_core>(jcp));
+    } else if (mayiuse(cpu::x64::avx2)) {
+        roi_align_kernel_.reset(new jit_uni_roi_align_kernel_f32<cpu::x64::avx2>(jcp));
+    } else if (mayiuse(cpu::x64::sse41)) {
+        roi_align_kernel_.reset(new jit_uni_roi_align_kernel_f32<cpu::x64::sse41>(jcp));
+    }
+
+    if (roi_align_kernel_)
+        roi_align_kernel_->create_ker();
 }
 
+namespace {
+struct ExperimentalDetectronROIFeatureExtractorContext {
+    ExperimentalDetectronROIFeatureExtractor &node;
+};
+}
+
+template<typename T>
+struct ExperimentalDetectronROIFeatureExtractor::ExperimentalDetectronROIFeatureExtractorExecute {
+    using srcT = typename std::tuple_element<0, T>::type;
+    using dstT = typename std::tuple_element<1, T>::type;
+
+    void operator()(ExperimentalDetectronROIFeatureExtractorContext& ctx) {
+        ctx.node.executeSpecified<srcT, dstT>();
+    }
+};
+
 void ExperimentalDetectronROIFeatureExtractor::execute(dnnl::stream strm) {
+    auto inputPrec = getParentEdgeAt(INPUT_FEATURES_START)->getMemory().GetDataType();
+    auto outputPrec = getChildEdgeAt(OUTPUT_ROI_FEATURES)->getMemory().GetDataType();
+    if (!((inputPrec == dnnl_bf16 && outputPrec == dnnl_bf16) ||
+          (inputPrec == dnnl_f32 && outputPrec == dnnl_f32)))
+        IE_THROW() <<"ExperimentalDetectronROIFeatureExtractor: precision is not supported";
+
+    ExperimentalDetectronROIFeatureExtractorContext ctx{*this};
+
+    OV_SWITCH(intel_cpu, ExperimentalDetectronROIFeatureExtractorExecute, ctx, std::tie(inputPrec, outputPrec),
+              OV_CASE2(dnnl_f32, dnnl_f32, float, float),
+              OV_CASE2(dnnl_bf16, dnnl_bf16, bfloat16_t, bfloat16_t))
+}
+
+template <typename inputType, typename outputType>
```

Comment:
I suggest to change `inputType` -> `InputType` and `outputType` -> `OutputType`

Reply:
Done